api/app/notifications/receive_notifications.py

```diff
@@ -11,132 +14,108 @@
 from app.dao.inbound_sms_dao import dao_create_inbound_sms
 from app.models import InboundSms, INBOUND_SMS_TYPE, SMS_TYPE
 from app.errors import register_errors
-from app.utils import convert_aet_to_utc
+from app.sap.oauth2 import require_oauth
 
 receive_notifications_blueprint = Blueprint('receive_notifications', __name__)
 register_errors(receive_notifications_blueprint)
 
 
-@receive_notifications_blueprint.route('/notifications/sms/receive/mmg', methods=['POST'])
-def receive_mmg_sms():
-    """
-    {
-        'MSISDN': '447123456789'
-        'Number': '40604',
-        'Message': 'some+uri+encoded+message%3A',
-        'ID': 'SOME-MMG-SPECIFIC-ID',
-        'DateRecieved': '2017-05-21+11%3A56%3A11'
-    }
-    """
-    post_data = request.get_json()
-
-    auth = request.authorization
+@receive_notifications_blueprint.route('/notifications/sms/receive/sap', methods=['POST'])
+@require_oauth(scope=None)
+def receive_sap_sms():
+    response = MessagingResponse()
 
-    if not auth:
-        current_app.logger.warning("Inbound sms (MMG) no auth header")
-        abort(401)
-    elif auth.username not in current_app.config['MMG_INBOUND_SMS_USERNAME'] \
-            or auth.password not in current_app.config['MMG_INBOUND_SMS_AUTH']:
-        current_app.logger.warning("Inbound sms (MMG) incorrect username ({}) or password".format(auth.username))
-        abort(403)
+    data = json.loads(request.data)
 
-    inbound_number = strip_leading_sixty_one(post_data['Number'])
+    service = fetch_potential_service(data['originatingAddress'], 'sap')
 
-    service = fetch_potential_service(inbound_number, 'mmg')
     if not service:
-        # since this is an issue with our service <-> number mapping, or no inbound_sms service permission
-        # we should still tell MMG that we received it successfully
-        return 'RECEIVED', 200
+        # Since this is an issue with our service <-> number mapping, or no
```

Comment:
we should do something here to log or alert ourselves that something is going on..

could we manually tell sentry to report an error?

https://docs.sentry.io/error-reporting/capturing/?platform=python#capturing-messages